Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,170 @@ This document includes four main sections:
+Call the :func:`sqlite3.connect` function to
+to create a connection to a database :file:`tutorial.db`
+in the current working directory,
+implicitly creating it if it does not exist::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+The returned :class:`Connection` object ``con``
+represents the connection to the database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+you'll use a database cursor.
+Call :meth:`con.cursor() <Connection.cursor>` to create the :class:`Cursor`::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
-
-   # Insert a row of data
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
+Now you'll create a database table ``movie`` with columns for title,
+release year, and review score.
+For simplicitly, just use column names in the table declaration:
+thanks to the `flexible typing`_ feature of SQLite,
+specifying the data types is optional.
+Execute the ``CREATE TABLE`` statement
+by calling :meth:`con.execute() <Cursor.execute>`::
+
+   cur.execute("CREATE TABLE movie(title, year, score)")
+
+.. Ideally, we'd use sqlite_schema instead of sqlite_master below,
+   but earlier versions of SQLite do not recognise that variant.
+
+You can verify that the new table has been created by querying
+the ``sqlite_master`` table built-in to SQLite.
+It should contain an entry for the ``movie`` table definition.
+Execute that query by calling :meth:`cur.execute(...) <Cursor.execute>`,
+store the result in a variable ``res``,
+and call :meth:`res.fetchone() <Cursor.fetchone>` to fetch the first
+(and only) row that was returned::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master")
+   >>> res.fetchone()
+   ('movie',)
+
+As expected, the query shows the table is now created
+by returning a single :class:`tuple`: with the name of the table.
+As an exercise, try querying ``sqlite_master``
+for a non-existent table ``"abc"``::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master WHERE name='abc'")
+   >>> res.fetchone()
+   >>>
+
+As expected, the query returns an empty result.
+
+Now, you'll add two rows of data supplied as SQL literals
+by executing an ``INSERT`` statement,
+once again by calling :meth:`cur.execute(...) <Cursor.execute>`::
+
+   cur.execute("""
+       INSERT INTO movie VALUES
+           ('Monty Python and the Holy Grail', 1975, 8.2),
+           ('And Now for Something Completely Different', 1971, 7.5)
+   """)
+
+The ``INSERT`` statement implicitly opens a transaction,
```

Comment:
Something that is not immediately obvious to me, and that it might be worth mentioning (here or elsewhere (e.g. in the transactions howto)), is that it appears that not all queries open a transaction.  In this tutorial the `CREATE` and the `SELECT` don't seem to open it, whereas the `INSERT` does.

After which queries do I need to commit?

Reply:
FYI, the second sentence of the Transaction Control section linked here states,

> If the connection attribute [isolation_level](https://docs.python.org/3.12/library/sqlite3.html#sqlite3.Connection.isolation_level) is not None, new transactions are implicitly opened before [execute()](https://docs.python.org/3.12/library/sqlite3.html#sqlite3.Cursor.execute) and [executemany()](https://docs.python.org/3.12/library/sqlite3.html#sqlite3.Cursor.executemany) executes INSERT, UPDATE, DELETE, or REPLACE statements.

I'm not sure whether its worth complicating the tutorial by explicitly enumerating those here

Reply:
If it's already mentioned in the linked section, then you can disregard my comment.